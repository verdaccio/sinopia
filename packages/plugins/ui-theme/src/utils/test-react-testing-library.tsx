import { render } from '@testing-library/react';
import React from 'react';
import { I18nextProvider } from 'react-i18next';
import { Provider } from 'react-redux';

import ThemeProvider from 'verdaccio-ui/design-tokens/ThemeProvider';
import APIProvider from 'verdaccio-ui/providers/API/APIProvider';
import AppConfigurationProvider from 'verdaccio-ui/providers/config';

import i18nConfig from '../i18n/config';

const renderWithStore = (ui, store) =>
  render(ui, {
    wrapper: ({ children }) => (
      <Provider store={store}>
        <AppConfigurationProvider>
          <APIProvider>
            <ThemeProvider>
              <I18nextProvider i18n={i18nConfig}>{children}</I18nextProvider>
            </ThemeProvider>
          </APIProvider>
        </AppConfigurationProvider>
      </Provider>
    ),
  });

const customRender = (node: React.ReactElement, ...options: any) => {
  return render(
    <AppConfigurationProvider>
      <APIProvider>
        <ThemeProvider>
          <I18nextProvider i18n={i18nConfig}>{node}</I18nextProvider>
        </ThemeProvider>
      </APIProvider>
    </AppConfigurationProvider>,
    ...options
  );
};

export * from '@testing-library/react';
// FIXME: rename all references with customRemder
export { customRender as render };
export { customRender, renderWithStore };
