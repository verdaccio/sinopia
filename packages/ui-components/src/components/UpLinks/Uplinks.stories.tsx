import React from 'react';

import { default as Uplinks } from '.';

export default {
  title: 'Components/Detail/Uplinks',
};

export const UplinksAll: any = () => {
  return (
    <Uplinks
      packageMeta={{
        latest: {
          name: 'verdaccio',
          version: '4.0.0',
        },
        _uplinks: {
          npmjs: {
            etag: '"W/"252f0a131cedd3ea82dfefd6fa049558""',
            fetched: 1529779934081,
          },
          yarn: {
            etag: '"W/"252f0a131cedd3ea82dfefd6fa049558""',
            fetched: 1529779934081,
          },
          private: {
            etag: '"W/"252f0a131cedd3ea82dfefd6fa049558""',
            fetched: 1529779934081,
          },
        },
      }}
    />
  );
};
export const NoUplinks: any = () => {
  return (
    <Uplinks
      packageMeta={{
        latest: {
          name: 'verdaccio',
          version: '4.0.0',
        },
        _uplinks: {},
      }}
    />
  );
};
