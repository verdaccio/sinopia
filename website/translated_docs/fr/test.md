---
id: unit-testing
title: "Test unitaire"
---
Tous les tests sont divisés en trois dossiers:

- `test/unit` - Tests couvrant les fonctions transformant les données de manière non triviale. Ces tests `demandent()` simplement peu de fichiers et exécutent le code qu'ils contiennent, ils sont donc très rapides.
- `test/fonctionnel` - Tests qui lancent une instance verdaccio et effectuent une série de requêtes via http. Ils sont plus lents que les tests unitaires.
- `test/integration` - Tests qui lancent une instance de verdaccio et lui adressent des requêtes à l'aide de npm. Ils sont considérablement lents et peuvent atteindre un registre npm réel. **test non maintenu**

Les tests unitaires et fonctionnels sont effectués automatiquement lorsque vous démarrez `npm test` à partir du dossier racine du projet. Les tests d'intégration doivent être effectués manuellement de temps à autre.

L'on utilise `jest` pour tous les tests.

## Le Script npm

Pour exécuter le script de test, vous pouvez utiliser soit `npm` ou `yarn`.

    yarn run test
    

Cela activera uniquement les deux premiers groupes de test, unité et fonctionnel.

### Utilisation de test/unit

Ce qui suit est juste un exemple de ce à quoi un test unitaire devrait ressembler. Suivez fondamentalement la norme `jest`.

Essayez de décrire ce que fait exactement le test unitaire en une seule phrase dans l'en-tête de la section `test`.

```javacript
const verdaccio = require('../../src/api/index');
const config = require('./partials/config');

describe('basic system test', () => {

  beforeAll(function(done) {
    // something important
  });

  afterAll((done) => {
    // undo something important
  });

  test('server should respond on /', done => {
    // your test
    done();
  });
});
```

### Utilisation de test/functional

Le test fonctionnel de verdaccio présente un niveau de complexité supérieur, qui nécessite une explication détaillée pour garantir une expérience positive.

Tout commence dans le fichier de`index.js`. Plongeons-nous dedans.

```javascript
// we create 3 server instances
 const config1 = new VerdaccioConfig(
    './store/test-storage',
    './store/config-1.yaml',
    'http://localhost:55551/');
  const config2 = new VerdaccioConfig(
      './store/test-storage2',
      './store/config-2.yaml',
      'http://localhost:55552/');
  const config3 = new VerdaccioConfig(
        './store/test-storage3',
        './store/config-3.yaml',
        'http://localhost:55553/');
  const server1: IServerBridge = new Server(config1.domainPath);
  const server2: IServerBridge = new Server(config2.domainPath);
  const server3: IServerBridge = new Server(config3.domainPath);
  const process1: IServerProcess = new VerdaccioProcess(config1, server1, SILENCE_LOG);
  const process2: IServerProcess = new VerdaccioProcess(config2, server2, SILENCE_LOG);
  const process3: IServerProcess = new VerdaccioProcess(config3, server3, SILENCE_LOG);
  const express: any = new ExpressServer();
  ...

    // we check whether all instances has been started, since run in independent processes
    beforeAll((done) => {
      Promise.all([
        process1.init(),
        process2.init(),
        process3.init()]).then((forks) => {
          _.map(forks, (fork) => {
            processRunning.push(fork[0]);
          });
          express.start(EXPRESS_PORT).then((app) =>{
            done();
          }, (err) => {
            done(err);
          });
      }).catch((error) => {
        done(error);
      });
    });

    // after finish all, we ensure are been stoped
    afterAll(() => {
      _.map(processRunning, (fork) => {
        fork.stop();
      });
      express.server.close();
    });


```

### Utilisation

Ici nous allons décrire à quoi devrait ressembler un test fonctionnel typique. Vérifiez inline pour plus d'informations détaillées.

#### The lib/server.js

La classe de serveur est uniquement un wrapper qui simule un client `npm` et fournit une simple API pour les tests fonctionnels.

Comme nous l'avons mentionné dans la section précédente, nous créons 3 serveurs de processus accessibles dans chaque processus, tels que `serveur1`, `serveur2` et `` serveur3`.

En utilisant ces références, vous pourrez envoyer des requêtes à chacune des 3 instances en cours d'exécution.

```javascript
<br />export default function(server) {
  // we recieve any server instance via arguments
  test('add tag - 404', () => {
    // we interact with the server instance.
    return server.addTag('testpkg-tag', 'tagtagtag', '0.0.1').status(404).body_error(/no such package/);
  });
});
```

### Test/integration

Cette section n'a jamais été utilisée, mais nous recherchons de l'aide pour la faire fonctionner correctement. **Toute nouvelle idée est la bienvenue.**