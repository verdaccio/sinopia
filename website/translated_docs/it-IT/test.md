---
id: unit-testing
title: "Test Unità"
---
Verdaccio si concentra molto sui test delle unità e sui test end to end. Se sei disposto a contribuire, fai riferimento alla seguente [guida per maggiori informazioni](https://github.com/verdaccio/verdaccio/wiki/Developing-new-tests).