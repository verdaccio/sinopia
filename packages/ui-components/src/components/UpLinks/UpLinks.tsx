import Card from '@mui/material/Card';
import CardContent from '@mui/material/CardContent';
import List from '@mui/material/List';
import ListItem from '@mui/material/ListItem';
import React from 'react';
import { useTranslation } from 'react-i18next';

import { utils } from '../../utils';
import NoItems from '../NoItems';
import { ListItemText, Spacer, StyledText } from './styles';

const UpLinks: React.FC<{ packageMeta: any }> = ({ packageMeta }) => {
  const { t } = useTranslation();

  if (!packageMeta?._uplinks || !packageMeta.latest) {
    return null;
  }

  const { _uplinks: uplinks, latest } = packageMeta;

  if (Object.keys(uplinks).length === 0) {
    return <NoItems data-testid="no-uplinks" text={t('uplinks.no-items', { name: latest.name })} />;
  }

  return (
    <Card>
      <CardContent>
        <StyledText variant="subtitle1">{t('uplinks.title')}</StyledText>
        <List>
          {Object.keys(uplinks)
            .reverse()
            .map((name) => (
              <ListItem key={name}>
                <ListItemText>{name}</ListItemText>
                <Spacer />
                <ListItemText>{utils.formatDateDistance(uplinks[name].fetched)}</ListItemText>
              </ListItem>
            ))}
        </List>
      </CardContent>
    </Card>
  );
};

export default UpLinks;
