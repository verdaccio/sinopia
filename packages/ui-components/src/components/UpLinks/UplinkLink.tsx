import React from 'react';

import { utils } from '../../utils';
import LinkExternal from '../LinkExternal';

const UpLinkLink: React.FC<{ packageName: string; uplinkName: string }> = ({
  packageName,
  uplinkName,
}) => {
  const link = utils.getUplink(uplinkName, packageName);
  return link ? (
    <LinkExternal to={link} variant="outline">
      {uplinkName}
    </LinkExternal>
  ) : (
    <>{uplinkName}</>
  );
};

export default UpLinkLink;
