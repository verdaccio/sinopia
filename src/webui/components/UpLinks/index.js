/**
 * @prettier
 */
import React from 'react';
import List from '@material-ui/core/List/index';
import ListItem from '@material-ui/core/ListItem/index';

import { DetailContextConsumer } from '../../pages/version/index';
import NoItems from '../NoItems';
import { formatDateDistance } from '../../utils/package';

import { Heading, Spacer, ListItemText } from './styles';

class UpLinks extends React.PureComponent<any> {
  render() {
    return (
      // $FlowFixMe
      <DetailContextConsumer>
        {({ packageMeta }) => {
          return this.renderContent(packageMeta._uplinks, packageMeta.latest);
        }}
      </DetailContextConsumer>
    );
  }

  renderUpLinksList = uplinks => (
    <List>
      {Object.keys(uplinks)
        .reverse()
        .map(name => (
          <ListItem key={name}>
            <ListItemText>{name}</ListItemText>
            <Spacer />
            <ListItemText>{`${formatDateDistance(uplinks[name].fetched)} ago`}</ListItemText>
          </ListItem>
        ))}
    </List>
  );

  renderContent(uplinks, { name }) {
    if (Object.keys(uplinks).length > 0) {
      return (
        uplinks && (
          <>
            <Heading variant={'subheading'}>{'Uplinks'}</Heading>
            {this.renderUpLinksList(uplinks)}
          </>
        )
      );
    }
    return <NoItems text={`${name} has no uplinks.`} />;
  }
}

export default UpLinks;
