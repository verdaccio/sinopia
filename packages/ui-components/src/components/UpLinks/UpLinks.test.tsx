import React from 'react';

import { cleanup, render } from '../../test/test-react-testing-library';
import UpLinks from './UpLinks';

describe('<UpLinks /> component', () => {
  beforeEach(cleanup);

  test('should render the component when there is no uplink', () => {
    const packageMeta = {
      latest: {
        name: 'verdaccio',
        version: '4.0.0',
        dist: { fileCount: 0, unpackedSize: 0 },
      },
      _uplinks: {},
    };

    const wrapper = render(<UpLinks packageMeta={packageMeta} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should render the component with uplinks', () => {
    const packageMeta = {
      latest: {
        name: 'verdaccio',
        version: '4.0.0',
        author: {
          name: 'verdaccio user',
          url: '',
          avatar: 'https://www.gravatar.com/avatar/000000',
        },
        dist: { fileCount: 0, unpackedSize: 0 },
      },
      _uplinks: {
        npmjs: {
          etag: '"W/"252f0a131cedd3ea82dfefd6fa049558""',
          fetched: 1529779934081,
        },
      },
    };

    const wrapper = render(<UpLinks packageMeta={packageMeta} />);

    expect(wrapper).toMatchSnapshot();
  });

  test('should not render if input is missing', () => {
    const wrapper = render(<UpLinks packageMeta={undefined} />);
    // expect nothing to be rendered
    expect(wrapper.queryByTestId('no-uplinks-npm')).toBeNull();
    expect(wrapper.queryByTestId('uplinks')).toBeNull();
  });
});
