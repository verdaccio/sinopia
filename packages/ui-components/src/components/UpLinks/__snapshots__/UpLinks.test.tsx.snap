// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UpLinks /> component should render the component when there is no uplink 1`] = `
{
  "asFragment": [Function],
  "baseElement": .emotion-0 {
  background-color: #fff;
  color: rgba(0, 0, 0, 0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: none;
  background-image: unset;
  font-family: -apple-system,BlinkMacSystemFont,"Helvetica Neue",Arial,sans-serif;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  background-color: rgb(229, 246, 253);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 6px 16px;
  color: rgb(1, 67, 97);
}

.emotion-0 .MuiAlert-icon {
  color: #0288d1;
}

.emotion-1 {
  margin-right: 12px;
  padding: 7px 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 22px;
  opacity: 0.9;
}

.emotion-2 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: inherit;
}

.emotion-3 {
  padding: 8px 0;
  min-width: 0;
  overflow: auto;
}

.emotion-4 {
  margin: 0;
  font-family: -apple-system,BlinkMacSystemFont,"Helvetica Neue",Arial,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
}

<body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardInfo MuiAlert-standard emotion-0"
        role="alert"
      >
        <div
          class="MuiAlert-icon emotion-1"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit emotion-2"
            data-testid="InfoOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20, 12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10, 10 0 0,0 12,2M11,17H13V11H11V17Z"
            />
          </svg>
        </div>
        <div
          class="MuiAlert-message emotion-3"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 emotion-4"
            data-testid="no-uplinks"
          >
            uplinks.no-items
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": .emotion-0 {
  background-color: #fff;
  color: rgba(0, 0, 0, 0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: none;
  background-image: unset;
  font-family: -apple-system,BlinkMacSystemFont,"Helvetica Neue",Arial,sans-serif;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  background-color: rgb(229, 246, 253);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 6px 16px;
  color: rgb(1, 67, 97);
}

.emotion-0 .MuiAlert-icon {
  color: #0288d1;
}

.emotion-1 {
  margin-right: 12px;
  padding: 7px 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 22px;
  opacity: 0.9;
}

.emotion-2 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: inherit;
}

.emotion-3 {
  padding: 8px 0;
  min-width: 0;
  overflow: auto;
}

.emotion-4 {
  margin: 0;
  font-family: -apple-system,BlinkMacSystemFont,"Helvetica Neue",Arial,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
}

<div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardInfo MuiAlert-standard emotion-0"
      role="alert"
    >
      <div
        class="MuiAlert-icon emotion-1"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit emotion-2"
          data-testid="InfoOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20, 12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10, 10 0 0,0 12,2M11,17H13V11H11V17Z"
          />
        </svg>
      </div>
      <div
        class="MuiAlert-message emotion-3"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 emotion-4"
          data-testid="no-uplinks"
        >
          uplinks.no-items
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<UpLinks /> component should render the component with uplinks 1`] = `
{
  "asFragment": [Function],
  "baseElement": .emotion-0 {
  background-color: #fff;
  color: rgba(0, 0, 0, 0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
  background-image: unset;
  overflow: hidden;
}

.emotion-1 {
  padding: 16px;
}

.emotion-1:last-child {
  padding-bottom: 24px;
}

.emotion-3 {
  margin: 0;
  font-family: -apple-system,BlinkMacSystemFont,"Helvetica Neue",Arial,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.75;
  font-weight: 700;
}

.emotion-4 {
  list-style: none;
  margin: 0;
  padding: 0;
  position: relative;
  padding-top: 8px;
  padding-bottom: 8px;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: 100%;
  box-sizing: border-box;
  text-align: left;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
}

.emotion-5.Mui-focusVisible {
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-5.Mui-selected {
  background-color: rgba(75, 94, 64, 0.08);
}

.emotion-5.Mui-selected.Mui-focusVisible {
  background-color: rgba(75, 94, 64, 0.2);
}

.emotion-5.Mui-disabled {
  opacity: 0.38;
}

.emotion-7 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  min-width: 0;
  margin-top: 4px;
  margin-bottom: 4px;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  color: #000;
  opacity: 0.6;
}

.emotion-8 {
  margin: 0;
  font-family: -apple-system,BlinkMacSystemFont,"Helvetica Neue",Arial,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  display: block;
}

.emotion-9 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  border-bottom: 1px dotted rgba(0, 0, 0, 0.2);
  white-space: nowrap;
  height: 0.5em;
}

<body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root emotion-0"
      >
        <div
          class="MuiCardContent-root emotion-1"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle1 emotion-2 emotion-3"
          >
            uplinks.title
          </h6>
          <ul
            class="MuiList-root MuiList-padding emotion-4"
          >
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding emotion-5"
            >
              <div
                class="MuiListItemText-root emotion-6 emotion-7"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary emotion-8"
                >
                  npmjs
                </span>
              </div>
              <div
                class="emotion-9 emotion-10"
              />
              <div
                class="MuiListItemText-root emotion-6 emotion-7"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary emotion-8"
                >
                  6 years ago
                </span>
              </div>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </body>,
  "container": .emotion-0 {
  background-color: #fff;
  color: rgba(0, 0, 0, 0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
  background-image: unset;
  overflow: hidden;
}

.emotion-1 {
  padding: 16px;
}

.emotion-1:last-child {
  padding-bottom: 24px;
}

.emotion-3 {
  margin: 0;
  font-family: -apple-system,BlinkMacSystemFont,"Helvetica Neue",Arial,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.75;
  font-weight: 700;
}

.emotion-4 {
  list-style: none;
  margin: 0;
  padding: 0;
  position: relative;
  padding-top: 8px;
  padding-bottom: 8px;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: 100%;
  box-sizing: border-box;
  text-align: left;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
}

.emotion-5.Mui-focusVisible {
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-5.Mui-selected {
  background-color: rgba(75, 94, 64, 0.08);
}

.emotion-5.Mui-selected.Mui-focusVisible {
  background-color: rgba(75, 94, 64, 0.2);
}

.emotion-5.Mui-disabled {
  opacity: 0.38;
}

.emotion-7 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  min-width: 0;
  margin-top: 4px;
  margin-bottom: 4px;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  color: #000;
  opacity: 0.6;
}

.emotion-8 {
  margin: 0;
  font-family: -apple-system,BlinkMacSystemFont,"Helvetica Neue",Arial,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  display: block;
}

.emotion-9 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  border-bottom: 1px dotted rgba(0, 0, 0, 0.2);
  white-space: nowrap;
  height: 0.5em;
}

<div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root emotion-0"
    >
      <div
        class="MuiCardContent-root emotion-1"
      >
        <h6
          class="MuiTypography-root MuiTypography-subtitle1 emotion-2 emotion-3"
        >
          uplinks.title
        </h6>
        <ul
          class="MuiList-root MuiList-padding emotion-4"
        >
          <li
            class="MuiListItem-root MuiListItem-gutters MuiListItem-padding emotion-5"
          >
            <div
              class="MuiListItemText-root emotion-6 emotion-7"
            >
              <span
                class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary emotion-8"
              >
                npmjs
              </span>
            </div>
            <div
              class="emotion-9 emotion-10"
            />
            <div
              class="MuiListItemText-root emotion-6 emotion-7"
            >
              <span
                class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary emotion-8"
              >
                6 years ago
              </span>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
